utils/frame/remote-externalities/src/lib.rs

```diff
@@ -351,14 +351,57 @@ where
 		prefix: StorageKey,
 		at: B::Hash,
 	) -> Result<Vec<StorageKey>, &'static str> {
-		let mut last_key: Option<StorageKey> = None;
+		const NUM_PARTS: usize = 4; // number of parts to divide the prefix into
+
+		let mut tasks = Vec::with_capacity(NUM_PARTS);
+
+		// Divide the prefix into N parts
+		let mut start_key = prefix.clone();
+		let mut end_key = prefix.clone();
 		let mut all_keys: Vec<StorageKey> = vec![];
-		let keys = loop {
+
+		for _ in 0..NUM_PARTS {
+			end_key.0[0] = start_key.0[0].wrapping_add(1);
+			let task = self.rpc_get_keys_paged_range(start_key.clone(), end_key.clone(), at);
+			tasks.push(task);
+			start_key.0[0] = end_key.0[0].wrapping_add(1);
+		}
```

Comment:
it's not clear to me how this splits the prefix into N equal sized parts and appears to be buggy with the 'index out of bounds' error.

to make this code more solid, I suggest:
1. abstract the prefix splitting code into a dedicated function
2. add comments into the function clearly explaining how it works (including examples in the comments if you can) and 
3. add a test for the function making sure it handles all regular and edge cases you can think of

Reply:
I've abstracted the prefix splitting code in a separate function named `split_storage_key_prefix`  and have also added comments explaining how it works.